connection/connection.go

```diff
@@ -0,0 +1,69 @@
+package connection
+
+import (
+	"context"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/kubernetes-csi/csi-lib-utils/protosanitizer"
+	"google.golang.org/grpc"
+)
+
+const (
+	// Interval of logging connection errors
+	connectionLoggingInterval = 10 * time.Second
+)
+
+// Connect opens insecure gRPC connection to a CSI driver. Address must have either '<protocol>://' prefix, or be
+// a path to a socket file. The function tries to connect indefinitely every second until it connects.
+// The function automatically adds interceptor for gRPC message logging.
+func Connect(address string, dialOptions ...grpc.DialOption) (*grpc.ClientConn, error) {
+	dialOptions = append(dialOptions,
+		grpc.WithInsecure(),                   // Don't use TLS, it's usually local Unix domain socket in a container.
+		grpc.WithBackoffMaxDelay(time.Second), // Retry every second after failure.
+		grpc.WithBlock(),                      // Block until connection succeeds.
+		grpc.WithUnaryInterceptor(LogGRPC),    // Log all messages.
+	)
+	if strings.HasPrefix(address, "/") {
```

Comment:
This doesn't handle an explicit `unix:///` prefix.

Reply:
`Dial()` handles `unix:///` just fine without any special dialer.

Reply:
We should move all repos to klog in one (short) burst. Is it the right time?

Reply:
Are you sure? https://github.com/grpc/grpc-go/issues/1741 was asking for this and then the feature request moved to https://github.com/grpc/grpc-go/issues/1911, which is still open.

I was under the impression that the custom dialer was used specifically to work around that shortcoming in gRPC. Perhaps I misunderstood. :man_shrugging: 

Looks like a good test case, don't you think? :wink:

Reply:
It does indeed work. Then I don't understand what those open gRPC issues are about. It looks like the custom dialer is needed because an absolute path without the `unix:///` prefix will not be treated as a Unix domain socket.

I'm still a bit wary about not always using that dialer, though. It creates two different code paths in our `Connect` function which (theoretically) could have different behavior. In practice I've not found any relevant difference, but the slow tests in PR https://github.com/jsafrane/csi-lib-utils/pull/1 now only cover the case with an absolute path. I only tried the `unix:///` variant of those by editing the test source code.

Reply:
I added code that removes custom dialer and uses unix:// for all sockets. This changes behavior of existing sidecar containers that used custom dialer in old releases, IMO it should be fine.